### PR TITLE
Implement Union All

### PR DESCRIPTION
**Pull request by pstutz**
Co-Authored-By: Martin Junghanns <s1ck@users.noreply.github.com>

### CODE REVIEW

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala [hunk 1]
```diff
@@ -208,7 +209,8 @@ final case class ConstructGraph(
     implicit val session = prev.records.caps
     val inputTable = prev.records
 
-    val entityTable = createEntities(constructItems, inputTable)
+    val newEntityTag = if (initialSchema.tags.isEmpty) 0 else initialSchema.tags.max + 1
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala]: I see that this is created outside of CAPSPatternGraph now. However, I don't see the line where we put it back in the resulting schema? Am I missing something?

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala [hunk 2]
```diff
@@ -296,7 +305,7 @@ final case class ConstructGraph(
       case None => Set.empty[(SlotContent, Column)]
     }
 
-    labelTuples ++ propertyTuples + (OpaqueField(node.v) -> generateId(columnIdPartition, numberOfColumnPartitions))
+    labelTuples ++ propertyTuples + (OpaqueField(node.v) -> generateId(columnIdPartition, numberOfColumnPartitions).setTag(newEntityTag))
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala]: Nice implicit

File: okapi-api/src/main/scala/org/opencypher/okapi/impl/schema/TagSupport.scala [hunk 3]
```diff
@@ -43,6 +47,20 @@ trait TagSupport {
 
 object TagSupport {
 
+  implicit class TaggedSchema(s: Schema) {
```
Comment (s1ck) [inline okapi-api/src/main/scala/org/opencypher/okapi/impl/schema/TagSupport.scala]: nice

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala [hunk 4]
```diff
@@ -64,7 +64,11 @@ object DataFrameOps {
 
     def getTag: Int = {
       // TODO: Verify that the tag actually fits into an Int or by requiring and checking a minimum size of 32 bits for idBits when reading it from config
-      ((l & tagMask) >> idBits).toInt
+      ((l & tagMask) >>> idBits).toInt
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala]: Is there a corresponding Column function in Spark?

File: spark-cypher/src/test/scala/org/opencypher/spark/impl/DataFrameOpsTest.scala [hunk 5]
```diff
@@ -88,4 +87,43 @@ class TagSupportTest extends CAPSTestSuite with Matchers {
     idRows.toList.map(_.get(0).asInstanceOf[Long]).sorted shouldBe List(2, 2, 3, 3)
   }
 
+  def toTag(n: Int) = (n & Int.MaxValue) % 1000
+  def toId(n: Int) = (n & Int.MaxValue) >>> tagBits
+
+  it("sets tags - Scala") {
+    forAll { (n1: Int, n2: Int) =>
```
Comment (s1ck) [inline spark-cypher/src/test/scala/org/opencypher/spark/impl/DataFrameOpsTest.scala]: where are the input values being set?

File: okapi-api/src/test/scala/org/opencypher/okapi/impl/schema/TagSupportTest.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package org.opencypher.okapi.impl.schema
+
+import org.opencypher.okapi.api.schema.Schema
+import TagSupport._
+import org.opencypher.okapi.test.BaseTestSuite
+
+class TagSupportTest extends BaseTestSuite {
```
Comment (s1ck) [inline okapi-api/src/test/scala/org/opencypher/okapi/impl/schema/TagSupportTest.scala]: Nice tests!

**Comment by s1ck** [comment]
@DarthMax suggested to put `with TagSupport` on `CAPSSchema` and not `SchemaImpl` since the latter is being used by all implementations (and thus the `TagSupport` methods could also live on the `Schema` trait). wdyt?

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala (hunk 1 above)
Comment (pstutz) [reply spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala]: It's done here: https://github.com/opencypher/cypher-for-apache-spark/pull/326/files/718449f49b9e59bcb270962725576dc10acbd187..364e39cf9f09fc3c38f0de502c38f3dd5bc1dee9#diff-e146dfa2a31a205d7135345c166a8185R222

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala (hunk 2 above)
Comment (pstutz) [reply spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/UnaryOperators.scala]: Thanks. It feels a bit redundant with the trait though.

File: spark-cypher/src/test/scala/org/opencypher/spark/impl/DataFrameOpsTest.scala (hunk 5 above)
Comment (pstutz) [reply spark-cypher/src/test/scala/org/opencypher/spark/impl/DataFrameOpsTest.scala]: They're fed into the function by ScalaCheck https://www.scalacheck.org

**Pull request merged**